Generalize to PostCSS plugins and add support for brotli compression.

chriseppstein opened the pull request:
I'm working on some CSS compression tools and this looked like it could be generalized to support arbitrary postcss pipelines so I worked that up. The cli tool now takes a new required option to a small js file that does the actual css processing.

Also, brotli is becoming increasingly common so it now reports on brotli compression in addition to gzip. I also threw in uncompressed sizes for shit and giggles.

I've moved cssnano to a dev dependency. There's some example processors in the processors folder (also used for testing the cli).

ben-eb commented on package.json:
```diff
@@ -1,7 +1,7 @@
   "name": "css-size",
-  "version": "1.1.0",
-  "description": "Display the size of a CSS file.",
+  "version": "2.0.0",
```
Can you remove the version bump please? I use https://github.com/sindresorhus/np which does this along with git tags etc. 😄

chriseppstein replied:
This was my not-so-subtle way of noting that I'd broken backwards compatibility for some minor parts of the API. I've reverted the version bump.

chriseppstein commented:
To make cssnano the default I think it has to be moved to a runtime dependency and it's a lot of code to deliver into projects that aren't using it. Do you know if there's a way to make this the default without moving the dependency?

ben-eb commented:
I don't think there's a way without moving it into runtime dependencies. But like https://github.com/lukekarrys/js-size does with UglifyJS my original aim of the module was to show the size for compressed CSS, so I would prefer to leave it in as a runtime dependency.

I'm working on the size of the module for the next major, which will require less code due to the "safe by default" approach (no autoprefixer or modules such as postcss-reduce-idents) & https://github.com/ben-eb/caniuse-lite.

chriseppstein commented:
Alright. I'll work these changes up this morning. Thanks!

chriseppstein commented on processors/noop.js:
```diff
@@ -0,0 +1,5 @@
+
+module.exports = function(css, opts) {
+  return postcss([]).process(css, opts);
+}
```
Great idea. It's sorted now and documentation is in alignment.

chriseppstein commented on usage.txt:
```diff
@@ -4,8 +4,23 @@ Options:
-    --options           Pass through options to cssnano. For example,
+    --processor,  -p    Path to the postcss processor code.
+
+    --options           Pass through options to the postcss plugin. For example,
```
Good catch. updated.

chriseppstein commented:
welp, it looks like the brotli library doesn't work on node 0.12 due to this recent change:
https://github.com/MayhemYDG/iltorb/commit/41b8b4e8a3a9d00e33d845fd2329e9bd988ad208

How important is it to you to continue to support 0.12? Seems like if we're bumping the major version it's ok to require a newer node and take 0.12 out of the build?

ben-eb commented:
Yeah, I'm happy with removing 0.12, only reason I didn't do it yet was because I was waiting to upgrade this module to cssnano 4 and releasing new versions all at once. v3 is still supporting 0.12. 😃

ben-eb commented on src/cli.js:
```diff
@@ -26,7 +31,11 @@ if (opts.version) {
-            table(buf, opts.options).then((results) => console.log(results));
+            let processor = nano.process.bind(nano);
+            if (opts.processor) {
+                processor = require(path.resolve(process.cwd(), opts.processor));
```
Can we move this to `src/index.js`? I think it would be good to default to cssnano in both the API & CLI. 😃

chriseppstein replied:
@ben-eb yeah, I assume that means the processor would be the last argument instead of the first then for those two methods so that it can be easily ommitted.

ben-eb replied:
Yeah, that makes sense. 👍

chriseppstein commented:
@ben-eb Alright, Hopefully it's merge-ready now. I sorted out the build and updated the API to allow processor to default to nano. I also noticed that with vanilla requires as was used in the example code in the README, this project didn't interoperate because it was producing an ES module so I added a shim file -- I think I got that right because the tests still work and they're importing the `main` file (which changed).
